src/pages/resultsView/plots/PlotsTabSelectionHistory.ts

```diff
@@ -0,0 +1,93 @@
+import { AxisMenuSelection, PlotsTabGeneOption, PlotsTabOption } from "./PlotsTab";
+import autobind from "autobind-decorator";
+
+type Selection = {
+    gene?: PlotsTabGeneOption | undefined,
+    geneSet?: PlotsTabOption | undefined,
+    source?: PlotsTabOption | undefined,
+    treatment?: PlotsTabOption | undefined,
+
+}
+
+type FieldUpdater = (option: any) => void;
+
+export default class PlotsTabSelectionHistory {
+    private horizontal: Map<string, Selection> = new Map();
+    private vertical: Map<string, Selection> = new Map();
+
+    @autobind
+    public updateHorizontalFromSelection(newSelection: AxisMenuSelection): void {
+        this.updateAxisWithSelection(this.horizontal, newSelection);
+    }
+    
+    @autobind
+    public updateVerticalFromSelection(newSelection: AxisMenuSelection): void {
+        this.updateAxisWithSelection(this.vertical, newSelection);
+    }
+
+    private updateAxisWithSelection(axis: Map<string, Selection>, newSelection: AxisMenuSelection): void {
+        if (newSelection.dataType !== undefined) {
+            let selectionToUpdate = axis.get(newSelection.dataType)
+
+            if (selectionToUpdate === undefined) {
+                selectionToUpdate = {};
+            }
+
+            axis.set(newSelection.dataType, this.updateSelection(selectionToUpdate, newSelection))
+        }
+    }
+
+    private updateSelection(selectionToUpdate: Selection, newSelection: AxisMenuSelection): Selection {
```

Comment:
super nitpick but since it doesnt use `this` it would be cool for this method to be static